bpf/lib/arp.h

```diff
@@ -19,13 +19,15 @@ struct arp_eth {
 
 /* Check if packet is ARP request for IP */
 static __always_inline int arp_check(struct ethhdr *eth, struct arphdr *arp,
-				     union macaddr *mac)
+				     struct arp_eth *arp_eth,
+				     union macaddr *mac, __be32 *ip)
```

Comment:
The idea is that a specific IP to respond to is optional. If no specific IP is provided, then behave as a global responder as before. A global responder is still used from-netdev on the flannel device I think, just at your comment below.

Reply:
But can't you do the same by passing the value? And if the IP is 0.0.0.0, consider that there is no IP to respond to?

Reply:
Yes, certainly, will do. It seemed more natural to me to use the pointer so I did not have to think whether 0.0.0.0 meant something else in this context. Just curious, are we worried here about dereferencing performance or something else?

Reply:
I don't feel strongly about it so you can keep it if you really prefer :) No worry about dereferencing, mostly I wanted to avoid to send the IP to the BPF stack if we can, and spare a couple BPF instructions.